### DISCUSSION TITLE
How to use lifetime in component definition？

### DISCUSSION THREAD

**Discussion by lisiur** [bevyengine/bevy#1691] [Q&A] [answered]
Suppose I have two structures A and B, A contains a tree
, B contains a reference to a node in A. They are all added to the system as resources, how to obtain and modify them in the system function?

Sample code:
```rust
struct A(Tree);
struct B(Option<&'static Node>);

fn main() {
    ...
   app.add_resource(A(tree));
   app.add_resouece(B(None));
}
```

**Comment by TheRawMeatball** [comment, marked as answer]
I don't think you can use lifetimes for this - all resources and components must be 'static. You should probably make a single 'static struct that wraps the, entire graph and put that in.
